cmd/prompt/prompt_test.go

```diff
@@ -0,0 +1,241 @@
+	"github.com/github/gh-models/pkg/command"
+	"github.com/github/gh-models/pkg/util"
+	"github.com/stretchr/testify/require"
+)
+
+func TestPrompt(t *testing.T) {
+	t.Run("NewPromptCommand happy path with model in prompt file", func(t *testing.T) {
+		client := azuremodels.NewMockClient()
+		fakeMessageFromModel := "Here's a summary of the text in bullet points:\n\n- Point 1\n- Point 2"
+		chatChoice := azuremodels.ChatChoice{
+			Message: &azuremodels.ChatChoiceMessage{
+				Content: util.Ptr(fakeMessageFromModel),
+				Role:    util.Ptr(string(azuremodels.ChatMessageRoleAssistant)),
+			},
+		}
+		chatCompletion := azuremodels.ChatCompletion{Choices: []azuremodels.ChatChoice{chatChoice}}
+		chatResp := &azuremodels.ChatCompletionResponse{
+			Reader: sse.NewMockEventReader([]azuremodels.ChatCompletion{chatCompletion}),
+		}
+		getChatCompletionCallCount := 0
+		client.MockGetChatCompletionStream = func(ctx context.Context, opt azuremodels.ChatCompletionOptions) (*azuremodels.ChatCompletionResponse, error) {
+			getChatCompletionCallCount++
+
+			// Verify that the request contains the expected data
+			require.Equal(t, "gpt-4o-mini", opt.Model)
+
+			// Find the user message and verify it contains our test content
+			var userMsgFound bool
+			var systemMsgFound bool
+			for _, msg := range opt.Messages {
+				if msg.Role == azuremodels.ChatMessageRoleUser && *msg.Content == "Summarize the given text:\n\n<text>\nTest content to summarize\n</text>" {
+					userMsgFound = true
+				}
+				if msg.Role == azuremodels.ChatMessageRoleSystem && *msg.Content == "You are a text summarizer. Your only job is to summarize a given text to you. I want you to summarize in bullet points." {
+					systemMsgFound = true
+				}
+			}
+			require.True(t, userMsgFound, "User message with correct content not found")
+			require.True(t, systemMsgFound, "System message with correct content not found")
+
+			return chatResp, nil
+		}
+
+		// Create a temp prompt file
+		promptContent := `---
+name: Summarizer
+description: Summarizes a given text
+model: gpt-4o-mini
+model_parameters:
+  temperature: 0.5
+---
+
+system:
+You are a text summarizer. Your only job is to summarize a given text to you. I want you to summarize in bullet points.
+
+user:
+Summarize the given text:
+
+<text>
+{{text}}
+</text>
+`
+		tmpDir := t.TempDir()
+		promptFilePath := filepath.Join(tmpDir, "test-prompt.md")
+		err := os.WriteFile(promptFilePath, []byte(promptContent), 0644)
+		require.NoError(t, err)
+
+		buf := new(bytes.Buffer)
+		cfg := command.NewConfig(buf, buf, client, true, 80)
+		promptCmd := NewPromptCommand(cfg)
+		promptCmd.SetArgs([]string{promptFilePath, "text=Test content to summarize"})
+
+		_, err = promptCmd.ExecuteC()
+
+		require.NoError(t, err)
+		require.Equal(t, 1, getChatCompletionCallCount)
+		output := buf.String()
+		require.Contains(t, output, fakeMessageFromModel)
+	})
+
+	t.Run("NewPromptCommand error on missing prompt file", func(t *testing.T) {
```

Comment:
Can we add a test for when the prompt file name is specified but it doesn't exist?